include/caf/sensor_sampler.rst

```diff
@@ -0,0 +1,220 @@
+.. _caf_sensor_sampler:
+
+CAF: Sensor sampler module
+##########################
+
+.. contents::
+   :local:
+   :depth: 2
+
+The |sensor_sampler| of the :ref:`lib_caf` (CAF) generates the following types of events in relation with the sensor defined in the module configuration:
+
+* ``sensor_event`` when the sensor is sampled.
+* ``sensor_state_event`` when the sensor state changes.
+
+Configuration
+*************
+
+To use the module, you must complete the following requirements:
+
+1. Physically connect the device and add and enable it in the :file:`BOARD.dts` devicetree file.
+   For more information about adding sensor device to devicetree, refer to :ref:`zephyr:use-dt-overlays`.
+#. Enable the following Kconfig options:
+
+   * :option:`CONFIG_CAF_SENSOR_SAMPLER` - This option enables the |sensor_sampler|.
+   * :option:`CONFIG_SENSOR` - This option enables Zephyr's :ref:`zephyr:sensor_api` driver, which is required for interacting with the sensors.
+
+Additionally, you need to configure the sensor that you want to use in your application and enable it in the sensor Kconfig option.
+
+The following Kconfig options are also available for the module:
+
+* :option:`CONFIG_CAF_SENSOR_SAMPLER_DEF_PATH`
+* :option:`CONFIG_CAF_SENSOR_SAMPLER_THREAD_STACK_SIZE`
+* :option:`CONFIG_CAF_SENSOR_SAMPLER_THREAD_PRIORITY`
+
+Adding module configuration file
+================================
+
+In addition to setting the Kconfig options, you must also add a module configuration file that contains an array of :c:struct:`sensor_config`.
+
+To do so, complete the following steps:
+
+1. Add a file that defines the following information for every sensor that should be handled by the |sensor_sampler| in an array of :c:struct:`sensor_config`:
+
+   * :c:member:`sensor_config.dev_name` - Sensor device name.
+     The name must match the sensor label in the :file:`BOARD.dts` file.
+   * :c:member:`sensor_config.event_descr` - Sensor event description.
+     The sensor sampler submits this information every time the sensor is sampled.
```

Comment:
Maybe `The event description is used to identify sensor in application`.
The information is used also for sensor_state_events